File: CF_Quick-One/890_div2/B.py
from sys import stdin
def input(): return stdin.readline()[:-1]

def solve():


    N = int(input())
    A = list(map(int, input().split()))
    A = [a-1 for a in A]
    if N == 1:
        print("NO")
        return
    
    alpha = 0
    c = A.count(0)
    for i in A:
        if i != 0:
            alpha += i
        if alpha >= c:
            print("YES")
            return
    print("NO")
    

for _ in range(int(input())):
    solve()

File: CF_Quick-One/890_div2/B.java

// import java.util.Arrays;
import java.util.Scanner;


public class soln {

  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) {

    int t = scanner.nextInt();
    while (t-- > 0) {
      solve();
    }
    scanner.close();

  }

  public static void solve() {
    int N = scanner.nextInt();
    int A[] = new int[N];
    for (int i = 0; i < N; i++) {
      A[i] = scanner.nextInt() - 1;
    }
  
    if (N == 1){
      System.out.println("NO");
      return;
    }
    int c = 0;
    // count number of 0s
    for (int i = 0; i < N; i++) {
      if (A[i] == 0) {
        c++;
      }
    }

    int alpha = 0;
    for (int i : A) {
      alpha += i;
      if (alpha >= c){
        System.out.println("YES");
        return;
      }
    }
    System.out.println("NO");

  }
}
